python/cudf/cudf/core/abc.py

```diff
@@ -132,18 +122,10 @@ def device_deserialize(cls, header, frames):
         """
         typ = pickle.loads(header["type-serialized"])
         frames = [
-            cudf.core.buffer.as_device_buffer_like(f) if c else memoryview(f)
+            cudf.core.buffer.as_buffer(f) if c else memoryview(f)
             for c, f in zip(header["is-cuda"], frames)
         ]
-        assert all(
-            (type(f._owner) is rmm.DeviceBuffer)
-            if c
-            else (type(f) is memoryview)
-            for c, f in zip(header["is-cuda"], frames)
-        )
-        obj = typ.deserialize(header, frames)
-
-        return obj
+        return typ.deserialize(header, frames)
```

Comment:
Is removing the assertion just cleanup?

Reply:
Yes, and we do not enforce `type(f._owner) is rmm.DeviceBuffer` anywhere.